File: problems/problem34/problem34.py
# helpers
def Factorial(n):
    if n == 0 or n == 1:
        return 1
    
    Fac = 1
    
    for i in range(1, n + 1):
        Fac *= i
        
    return Fac
  
# main
# unoptimized
def Problem34(n):
    ProblemList = []
    
    for i in range(10, n + 1):
        iList = [int(j) for j in list(str(i))]
        
        FactorialSum = sum([Factorial(k) for k in iList])
        
        if FactorialSum == i:
            ProblemList.append(i)
            
    return sum(ProblemList)
  
  Problem34(int(1e6))

File: problems/problem34/Problem34.cpp
#include <iostream>
#include "Helpers.h"

int Problem34(int n)
{
    int Total = 0;

    for (int i = 10; i <= n; i++)
    {
        int FacSum = 0;

        for (char Digit : std::to_string(i))
        {
            FacSum += Helpers::Factorial((int)Digit - 48); // Use ASCII + typecasting to our advantage
        }

        if (FacSum == i)
        {
            Total += i;
        }
    }
    return Total;
}

int main()
{
    std::cout << Problem34(100000);
}
